sections/header.liquid

```diff
@@ -11,7 +11,7 @@
   <link rel="stylesheet" href="{{ 'component-mega-menu.css' | asset_url }}" media="print" onload="this.media='all'">
   <noscript>{{ 'component-mega-menu.css' | asset_url | stylesheet_tag }}</noscript>
 {%- endif -%}
-{%- if section.settings.enable_country_selector or section.settings.enable_language_selector -%}
+{%- if localization.available_countries.size > 1 or localization.available_languages.size > 1 -%}
```

Comment:
This allows for the language and currency pickers to be present in the mobile-menu drawers even when the the currency and language selectors on the header is unchecked

Reply:
Uploading Screen Recording 2023-07-05 at 8.30.26 AM.mov…

Reply:
https://github.com/Shopify/dawn/assets/93098869/469edc38-a87b-4a9e-906e-507a69a15bef

Reply:
I think then it means the CSS will be loaded anytime there is more than 1 language or currency. So we could then remove the call for this asset in the `footer.liquid` and `announcement-bar.liquid` 🤔 

And for the script `localization-form.js` it's loaded based on the activation of settings. So the issue here is that if you have the locales not checked in the announcement bar, header and footer, it won't load the script yet it will show the locales selectors in the drawer 😓

Reply:
should I keep it as it was? Or would there be a better way to do this?

Reply:
Is the footer related to the header and announcement bar in terms of the localization form? I am just trying to gain context on how changing how the script and styling is loaded in the header and announcement bar will after the footer section

Reply:
So I'd say, if we're always display language and currency picker in the drawer whenever there is more than 1 option. We should just load both the CSS and Script based on that condition in `theme.liquid` 🤔 

So I'd have: 
```liquid
{%- if localization.available_countries.size > 1 or localization.available_languages.size > 1 -%}
  {{ 'component-localization-form.css' | asset_url | stylesheet_tag: preload: true }}
  <script src="{{ 'localization-form.js' | asset_url }}" defer="defer"></script>
{%- endif -%}
```

The one little thing with that approach is that if it's only in the footer and menu drawer, we don't actually need the `preload: true` 🤔 

The script is all good. It's loaded everywhere the same way. 

What's your take @kmeleta ?